scripts/tool_discovery_benchmark/run_benchmark.py

```
@@ -0,0 +1,610 @@
+            for task in tasks:
+                group = await run_task_group(
+                    live_qa, args.binary, args.output_dir, arm, tool_count,
+                    task, args.repetitions,
+                )
+                for observation in group:
+                    observations.append(observation)
+                    with observations_path.open("a", encoding="utf-8") as handle:
+                        handle.write(json.dumps(observation, sort_keys=True) + "\n")
+    summary = {
+        "schema_version": 1,
+        "head": os.popen("git rev-parse HEAD").read().strip(),
```

Ruff (0.16.1)

[warning] Async functions should not create subprocesses with blocking methods

(ASYNC220)